src/gc/unix/cgroup.cpp

```diff
@@ -84,9 +66,113 @@ class CGroup
         free(mem_limit_filename);
         return result;
     }
+
+    bool GetCpuLimit(uint32_t *val)
+    {
+        char *cfs_period_filename = nullptr;
+        char *cfs_quota_filename = nullptr;
+        bool result = false;
+        long long quota;
+        long long period;
+        long long cpu_count;
+
+        if (m_cpu_cgroup_path == nullptr)
+            return false; 
+
+        cfs_period_filename = (char*)malloc(strlen(m_cpu_cgroup_path) + strlen(CFS_PERIOD_FILENAME) + 1);
+        if (cfs_period_filename == nullptr)
+            return false;
+
+        strcpy(cfs_period_filename, m_cpu_cgroup_path);
+        strcat(cfs_period_filename, CFS_PERIOD_FILENAME);
+        result = ReadLongLongValueFromFile(cfs_period_filename, &period);
+        free(cfs_period_filename);
+        if (!result || period <= 0)
+             return false;
```

Comment:
It would be nice to extract the lines 82..91 into a separate function and using it for both the period and quota instead of repeating the same block of code twice. You could make it so that the return value would be either positive for success and negative for failure and the function would get the CFS_PERIOD_FILENAME or CFS_QUOTA_FILENAME as a parameter.